ivy_tests/test_ivy/test_functional/test_core/test_statistical.py

```diff
@@ -82,7 +82,10 @@ def _statistical_dtype_values(draw, *, function, min_value=None, max_value=None)
                 | helpers.floats(min_value=0, max_value=max_correction - 1)
             )
         return dtype, values, axis, correction
-    return dtype, values, axis
+    dtype3, where = draw(
+        helpers.dtype_and_values(available_dtypes=["bool"], shape=shape)
```

Comment:
Another point, probably we can generate the `where` argument to have all broadcastable shapes of `shape` instead of `shape` directly

Reply:
@vedpatwardhan Could you pls specify the actual code to do this as I am unaware of this method and not able to get from the docs of Ivy.

Reply:
basically you could use something like [broadcastable_arrays](https://github.com/unifyai/ivy/blob/6e1f438cbe6d3cf7077f805648920074a8f15792/ivy_tests/test_ivy/test_functional/test_core/test_dtype.py#L75) to generate a boolean array for `where` along with the `values` if that makes sense?

Reply:
Apart from this, rest seems good to merge. Thanks @Aaryan562 😄

Reply:
> basically you could use something like [broadcastable_arrays](https://github.com/unifyai/ivy/blob/6e1f438cbe6d3cf7077f805648920074a8f15792/ivy_tests/test_ivy/test_functional/test_core/test_dtype.py#L75) to generate a boolean array for `where` along with the `values` if that makes sense?


@vedpatwardhan I'm sorry, but I didn't quite grasp that.😅 There is no explanation present to use it so I am finding it hard to decode the function and how to use it in my usecase.

Reply:
No worries, basically what we'd do is,
```
where_shape = helpers.mutually_broadcastable_shapes(num_shapes=1, base_shape=shape)
dtype3, where = draw(
        helpers.dtype_and_values(available_dtypes=["bool"], shape=where_shape)
)
```
Let me know if that works, thanks @Aaryan562 😄

Reply:
@vedpatwardhan Hey actually the above code did not work as we need to draw the values of `where_shape` like [this ](https://github.com/Aaryan562/ivy/blob/min_stat_arg/ivy_tests/test_ivy/test_frontends/test_numpy/test_broadcast/test_methods.py#L19). The code which worked was:
```
    where_shape = draw(helpers.mutually_broadcastable_shapes(num_shapes=1, base_shape=shape))
    dtype3, where = draw(
            helpers.dtype_and_values(available_dtypes=["bool"], shape=where_shape[0])
    )
```
The above code is creating errors as the `where_shape` sometimes generates more dimensions than the `axis` which thereby gives the error: 
```
E       ValueError: input operand has more dimensions than allowed by the axis remapping
```
The falsifying example is:
```
Trying example: test_min(
    backend_fw='numpy',
    on_device='cpu',
    dtype_and_x=(['uint32'],
     [array([ 2561, 32768,     0, 22144,  1537, 49163, 58670], dtype=uint32)],
     (0,),
     ['bool'],
     [array([[False]])]),
    keep_dims=False,
    initial=0,
    test_flags=FunctionTestFlags(
        ground_truth_backend='tensorflow',
        num_positional_args=1,
        with_out=False,
        with_copy=False,
        instance_method=False,
        test_gradients=False,
        test_trace=False,
        transpile=False,
        as_variable=[False],
        native_arrays=[True],
        container=[True],
        precision_mode=False,
    ),
    fn_name='min',
    ```

Reply:
Ah makes sense, probably we should generate both shapes from this helper, does passing `num_shapes=2` and then using one for `x` and the other for `where` work?

Reply:
@vedpatwardhan It generates different shape like [(1,), (3,)] which then gives 
```ValueError: non-broadcastable operand with shape (1,) doesn't match the broadcast shape (3,)```

Reply:
> @vedpatwardhan It generates different shape like [(1,), (3,)] which then gives `ValueError: non-broadcastable operand with shape (1,) doesn't match the broadcast shape (3,)`

@vedpatwardhan This was the msg which got overlooked.

Reply:
Oh sorry about missing this message, yeah let's just then skip using broadcasting here and revert back to what you were doing before where we used the same shape as `x` for generating the shape of `where`. Thanks @Aaryan562 😄

Reply:
> Oh sorry about missing this message, yeah let's just then skip using broadcasting here and revert back to what you were doing before where we used the same shape as `x` for generating the shape of `where`. Thanks @Aaryan562 😄

No worries, reverted the changes. Passed all the test cases successfully with 100 examples.

Reply:
sounds good!